5.5/root/usr/share/container-scripts/mysql/validate-variables.sh

```diff
@@ -32,11 +30,34 @@ function validate_variables() {
     root_specified=1
   fi
 
+  # If MYSQL_USER == "root", we have a special case
+  if [[ "${user_specified:-0}" == "1" && "$MYSQL_USER" == "root" ]]; then
+    if [[ "${root_specified:-0}" == "1" ]]; then
+      usage "You cannot set MYSQL_USER to 'root' and use MYSQL_PASSWORD and MYSQL_ROOT_PASSWORD at the same time"
+    fi
+    # We will now behave as if MYSQL_USER was not specified
+    export MYSQL_ROOT_PASSWORD="$MYSQL_PASSWORD"
+    export -n MYSQL_USER
+    export -n MYSQL_PASSWORD
+    user_specified=0
+    root_specified=1
+  fi
+
   # Either combination of user/pass/db or root password is ok
   if [[ "${user_specified:-0}" == "0" && "${root_specified:-0}" == "0" ]]; then
     usage
   fi
 
+  # If the root user is not specified, database name is required
```

Comment:
Because I've realized that we do not really need the database. You actually only need it in a replication setup, where you need to make sure it is listed in the configuration files so MySQL replicates it.I've gone through the code and the change should work, we don't use the variable anywhere else.